flash_attn/cute/flash_fwd_sm100.py

```diff
@@ -1907,3 +1958,113 @@ def make_and_init_load_kv_pipeline(self, load_kv_mbar_ptr):
     #     cute.arch.barrier_arrive(
     #         barrier_id=int(NamedBarrierFwd.WarpSchedulerWG1) + next_wg, number_of_threads=2 * 128,
     #     )
+
+    @cute.jit
+    def apply_score_mod(
+        self,
+        tSrS_t2r,
+        thr_tmem_load,
+        thr_mma_qk,
+        batch_idx,
+        head_idx,
+        m_block,
+        n_block,
+        softmax,
+        buffers = None,
+        fastdiv_mods = (None, None),
+        VEC_SIZE: cutlass.Constexpr[int] = 1,
```

Comment:
VEC_SIZE must = 1 for for current global load patterns, at least in inductor). I tried a number of different vec_sizes w/ no buffers and never found a perf benefit

Reply:
tried 2, 4, 8, perf caps at 2 for the 5 different examples I looked at. updates so that if we have buffers we drop back to 1 else we set to 2